File: Plugins/Aspose_Slides_Java_for_Ruby/lib/asposeslidesjava/Slides/accessslides.rb
module Asposeslidesjava
  module AccessSlides
    def initialize()
        # Using Slides Collection to Access Slide by Index
        get_slide_by_index()

        # Using Slides Collection to Access Slide by ID
        get_slide_by_id()
    end    

    def get_slide_by_index()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate Presentation class that represents the presentation file
        pres = Rjb::import('com.aspose.slides.Presentation').new(data_dir + 'Aspose.pptx')

        # Accessing a slide using its slide index
        slide = pres.getSlides().get_Item(0)

        puts "Slide: " + slide.to_string
    end    

    def get_slide_by_id()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate Presentation class that represents the presentation file
        pres = Rjb::import('com.aspose.slides.Presentation').new(data_dir + 'Aspose.pptx')

        # Getting Slide ID
        id = pres.getSlides().get_Item(0).getSlideId()
            
        # Accessing Slide by ID
        slide = pres.getSlideById(id)

        puts "Slide: " + slide.to_string
    end    
  end
end

File: Examples/src/main/java/com/aspose/slides/examples/Slides/AccessSlides.java
/* 
 * Copyright 2001-2013 Aspose Pty Ltd. All Rights Reserved.
 *
 * This file is part of Aspose.Slides. The source code in this file
 * is only intended as a supplement to the documentation, and is provided
 * "as is", without warranty of any kind, either expressed or implied.
 */
 
package com.aspose.slides.examples.Slides;

import com.aspose.slides.*;
import com.aspose.slides.examples.Utils;

public class AccessSlides
{
    public static void main(String[] args) throws Exception
    {
        // The path to the documents directory.
        String dataDir = Utils.getDataDir(AccessSlides.class);

        //Instantiate a Presentation object that represents a Presentation1 file
        Presentation pres = new Presentation(dataDir+ "demo.pptx");

        //Accessing a slide using its slide index
        ISlide slide = pres.getSlides().get_Item(0);

        System.out.println("Program executed successfully");
        
    }
}




